File: fixtures/scale/dart/large/lib/ciphers/rc4/rc4_suite.dart
import '../../core/cipher.dart';
import '../../core/cipher_suite.dart';
import '../../core/suite_result.dart';
import '../../verify/round_trip.dart';
import 'rc4_cipher.dart';
import 'rc4_key.dart';
import 'rc4_vectors.dart';

class Rc4Suite implements CipherSuite {
  @override
  String name() => 'rc4';

  @override
  String category() => 'cipher';

  @override
  SuiteResult run() {
    final Cipher cipher = Rc4Cipher(Rc4Key.defaultKey());
    var passed = 0;
    var failed = 0;
    final start = DateTime.now();
    for (final sample in rc4Samples()) {
      if (roundTripCheck(cipher, sample)) {
        passed++;
      } else {
        failed++;
      }
    }
    final elapsed = DateTime.now().difference(start).inMicroseconds;
    return SuiteResult(name(), passed, failed, elapsed);
  }
}

File: fixtures/scale/java/large/src/cryptobench/ciphers/rc4/Rc4Suite.java
package cryptobench.ciphers.rc4;

import cryptobench.core.Cipher;
import cryptobench.core.CipherSuite;
import cryptobench.core.SuiteResult;
import cryptobench.verify.RoundTrip;

public class Rc4Suite implements CipherSuite {
    @Override
    public String name() {
        return "rc4";
    }

    @Override
    public SuiteResult run() {
        Cipher cipher = new Rc4Cipher(Rc4Key.defaultKey());
        int passed = 0;
        int failed = 0;
        long start = System.nanoTime();
        for (String sample : Rc4Vectors.samples()) {
            if (RoundTrip.check(cipher, sample)) passed++; else failed++;
        }
        return new SuiteResult(name(), passed, failed, System.nanoTime() - start);
    }
}
